radix-engine/src/system/node.rs

```diff
@@ -66,15 +66,12 @@ pub enum RENodeInit {
 }
 
 impl RENodeInit {
-    pub fn to_substates(self) -> HashMap<SubstateOffset, RuntimeSubstate> {
-        let mut substates = HashMap::<SubstateOffset, RuntimeSubstate>::new();
+    pub fn to_substates(self) -> BTreeMap<SubstateOffset, RuntimeSubstate> {
```

Comment:
Why do we use different map for `RENodeInit` and not for `RENodeModuleInit`?

Reply:
I notice an opportunity to avoid the rewriting (from map to map) here, in `RENodeInit`.
Interestingly, I have missed the very same opportunity in `RENodeModuleInit` :D

Thanks for bringing this up - I now applied the same idea there 👍 
_(we are better off letting the `BTreeMap` stay here, because an instance is already available)_